Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/refine_mesh_at_isolevel.h

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) 2021-2023 GeometryFactory (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+//
+// Author(s)     : Sebastien Loriot
+
+#ifndef CGAL_POLYGON_MESH_PROCESSING_REFINE_MESH_AT_ISOLEVEL_H
+#define CGAL_POLYGON_MESH_PROCESSING_REFINE_MESH_AT_ISOLEVEL_H
+
+#include <CGAL/license/Polygon_mesh_processing/miscellaneous.h>
+
+#include <CGAL/boost/graph/named_params_helper.h>
+
+namespace CGAL {
+namespace Polygon_mesh_processing {
+
+/*!
+ * \ingroup PkgPolygonMeshProcessingRef
+ *
+ * refines `pm` by adding new vertices on edges having their incident vertices associated with
+ * values respectively larger and smaller than `isovalue` in `value_map`.
+ * The placement of new vertices on edges will be done by linear interpolation
+ * using the aforementioned values.
+ * New vertices will be associated `isovalue` in `value_map` when created.
+ * Additionally new edges will be added by connecting new vertices created sharing
+ * a common incident face. Note that in case more that two new vertices are added
+ * on a face boundary, no edges will be created in that face.
+ *
+ * @tparam PolygonMesh a model of the concepts `EdgeListGraph` and `FaceListGraph`
```

Comment:
Does it really work on a `PolygonMesh`, or should it be `TriangleMesh`.   Or is it the former, but as input you require it to be a triangle mesh, because the output will not be triangulated.

Reply:
It works on polygon meshes. As documented, the only restriction is to have at most two edges containing an intersection point with the isocurve. In such a case, no edges are added.

Reply:
Is that easy to report back to the user?

Reply:
I guess we could have a bool as return type of the function.

Reply:
What about an output iterator with the faces concerned?

Reply:
Might be a future small feature.